src/components/Catalog/Catalog.tsx

```diff
@@ -0,0 +1,81 @@
+import { JSX } from '@redneckz/uni-jsx';
+import type { UniBlockProps } from '../../types';
+import { ArrowButton } from '../../ui-kit/Button/ArrowButton';
+import { Description } from '../../ui-kit/Description/Description';
+import { Heading } from '../../ui-kit/Heading/Heading';
+import { CatalogCard } from './CatalogCard';
+import type { CatalogCardType, CatalogContent } from './CatalogContent';
+
+export interface CatalogProps extends CatalogContent, UniBlockProps {}
+
+const BLUR_BLOCK_CLASSES = 'absolute top-0 bottom-0 w-[84px]';
+const CARD_FULL_VIEW_COUNT = 3;
+const CARD_SHIFT = 430;
+
+export const Catalog = JSX<CatalogProps>(
+  ({ context, cards = [], className = '', title, description }) => {
+    const [activeCardIndex, setActiveCardIndex] = context.useState(0);
+
+    return (
+      <section className={`relative font-sans p-9 overflow-hidden text-center ${className}`}>
+        {title ? <Heading headingType="h2" className="mb-6" title={title} /> : null}
+        {description ? <Description className="mb-6" description={description} /> : null}
+        <div
+          className="flex duration-1000 gap-3.5"
+          style={{ transform: `translateX(-${activeCardIndex * CARD_SHIFT}px)` }}
+          role="list"
+        >
+          {cards?.length ? cards.map(renderCatalogCard({ context })) : null}
+        </div>
+        {renderNavButtons({
+          cardsCount: cards.length,
+          activeCardIndex,
+          setActiveCardIndex,
+        })}
+        <div className={`${BLUR_BLOCK_CLASSES} left-0 ${'bg-opacity-from-white'}`} />
+        <div className={`${BLUR_BLOCK_CLASSES} right-0 ${'bg-opacity-to-white'}`} />
+      </section>
+    );
+  },
+);
+
+const renderCatalogCard =
+  ({ context }) =>
```

Comment:
```suggestion
  (context) =>
```

Если сделать как выше что изменится? Чем обосновано помещение объекта сюда?
Ещё надо импортировать тип ```ContentPageContext``` и присвоить его.